lib/data_freshness/main.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python3
+
+import asyncio
+import os
+from datetime import datetime, timedelta
+from pngme.api import AsyncClient
+
+
+async def get_data_freshness(
+    api_client: AsyncClient,
+    user_uuid: str,
+    utc_starttime: datetime,
+    utc_endtime: datetime,
+    page: int,
+) -> float:
+    """Return the time in minutes between utc_endtime and the most recent financial event or alert,
+    as an indicator of data freshness.
+    Args:
+        api_client: Pngme Async API client
+        user_uuid: Pngme mobile phone user_uuid
+        utc_starttime: the time from which balances are considered
+        utc_endtime: the time until which balances are considered
+        page: the page(s) requested
+    Returns:
+        Return the time in minutes between utc_endtime and the most recent financial event or alert
+    """
+
+    # STEP 1: get lists of transactions, alerts, and balances for each institution
```

Comment:
This can just be `fetch list of institutions belonging to the user` 😄 as it's only hitting the `/institutions` api